src/list-sorting/index.js

```diff
@@ -1,3 +1,29 @@
-function listSorting(needle, haystack) {}
+"use strict";
+
+function listSorting(needle, haystack) {
+  let result;
+  let needleExist = false;
+
+
+  if(haystack.some(x => Array.isArray(x))){
```

Comment:
You can assign this condition a variable rather than put the function in the if statement.